include/Hdlcpp.hpp

```diff
@@ -35,27 +35,93 @@ namespace Hdlcpp {
 using TransportRead = std::function<int(std::span<uint8_t> buffer)>;
 using TransportWrite = std::function<int(const std::span<const uint8_t> buffer)>;
 
+template<typename T, std::size_t Capacity>
+class Buffer {
+    using Container = std::array<T, Capacity>;
+
+public:
+    bool empty() {
+        return std::span<typename Container::value_type>(m_head, m_tail).empty();
+    }
+
+    size_t capacity() {
+        return buffer.size();
+    }
+
+    typename Container::iterator begin() {
```

Comment:
I tried to keep this helper class as small as possible. Since `begin` and `end` are used for specifying `erase` and there for are needed to by manipulated.